tests: dma: s32z2xxdc2@D/s32z270/rtu1 : dma driver build failure

**Describe the bug**

A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - s32z2xxdc2@D/s32z270/rtu1
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - https://github.com/zephyrproject-rtos/zephyr/pull/80584


**To Reproduce**

west build -p -b s32z2xxdc2@D/s32z270/rtu1 tests/drivers/dma/chan_blen_transfer -T drivers.dma.chan_blen_transfer

**Expected behavior**

build pass

**Impact**

DMA functions

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:445:79: error: implicit declaration of function 'DMA_CSR_DREQ'; did you mean 'DMA_TCD_CSR_DREQ'? [-Werror=implicit-function-declaration]
  445 |                                         EDMA_TCD_CSR(tcd, kEDMA_EDMA4Flag) |= DMA_CSR_DREQ(1U);
      |                                                                               ^~~~~~~~~~~~
      |                                                                               DMA_TCD_CSR_DREQ
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c: In function 'dma_mcux_edma_reload':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:149:40: error: 'DMA_CSR_ACTIVE_MASK' undeclared (first use in this function); did you mean 'DMA_MP_CSR_ACTIVE_MASK'?
  149 | #define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CSR_ACTIVE_MASK)
      |                                        ^~~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:149:40: note: in definition of macro 'EDMA_HW_TCD_CH_ACTIVE_MASK'
  149 | #define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CSR_ACTIVE_MASK)
      |                                        ^~~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:149:40: note: each undeclared identifier is reported only once for each function it appears in
  149 | #define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CSR_ACTIVE_MASK)
      |                                        ^~~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/dma/dma_mcux_edma.c:149:40: note: in definition of macro 'EDMA_HW_TCD_CH_ACTIVE_MASK'
  149 | #define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CSR_ACTIVE_MASK)
      |                                        ^~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.

```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v4.0.0-rc2-202-g9c94ee53466c

--- a/drivers/dma/dma_mcux_edma.c
+++ b/drivers/dma/dma_mcux_edma.c
@@ -146,7 +146,12 @@ struct dma_mcux_edma_data {
 #define EDMA_TCD_CITER(tcd, flag)     ((tcd)->CITER)
 #define EDMA_TCD_CSR(tcd, flag)       ((tcd)->CSR)
 #define EDMA_TCD_DLAST_SGA(tcd, flag) ((tcd)->DLAST_SGA)
+#if defined(CONFIG_DMA_MCUX_EDMA_V3)
+#define DMA_CSR_DREQ                  DMA_TCD_CSR_DREQ
+#define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CH_CSR_ACTIVE_MASK)
+#else
 #define EDMA_HW_TCD_CH_ACTIVE_MASK    (DMA_CSR_ACTIVE_MASK)
+#endif /* CONFIG_DMA_MCUX_EDMA_V3 */
 #elif defined(CONFIG_DMA_MCUX_EDMA_V4)
 /* Above macros have been defined in fsl_edma_core.h */
 #define EDMA_HW_TCD_CH_ACTIVE_MASK (DMA_CH_CSR_ACTIVE_MASK)

